src/Routing/RouteCollection.php

```diff
@@ -148,6 +149,45 @@ public function parse($url, $method = '')
     }
 
     /**
+     * Takes the ServerRequestInterface, iterates the routes until one is able to parse the route.
+     *
+     * @param \Psr\Http\Messages\ServerRequestInterface $request The request to parse route data from.
+     * @return array An array of request parameters parsed from the URL.
+     * @throws \Cake\Routing\Exception\MissingRouteException When a URL has no matching route.
+     */
+    public function parseRequest(ServerRequestInterface $request)
+    {
+        $uri = $request->getUri();
+        $method = $request->getMethod();
+        $urlPath = $uri->getPath();
+        $host = $uri->getHost();
+        foreach (array_keys($this->_paths) as $path) {
+            if (strpos($urlPath, $path) !== 0) {
+                continue;
+            }
+
+            /* @var \Cake\Routing\Route\Route $route */
+            foreach ($this->_paths[$path] as $route) {
+                if (!$route->hostMatches($host)) {
```

Comment:
Why is the host check done here instead of inside `Route::parse()` where the method type and other checks are done?